test/Microsoft.ML.CpuMath.UnitTests.netcoreapp/UnitTests.cs

```diff
@@ -77,431 +81,661 @@ public CpuMathUtilsUnitTests()
             _testDstVectors = new AlignedArray[] { testDstVectorAligned1, testDstVectorAligned2 };
         }
 
+        private static void CheckProperFlag(string mode)
+        {
+#if NETCOREAPP30
+            if (mode == defaultMode)
+            {
+                Assert.True(System.Runtime.Intrinsics.X86.Avx.IsSupported);
+                Assert.True(System.Runtime.Intrinsics.X86.Sse.IsSupported);
+            }
+            else if (mode == disableAvx)
+            {
+                Assert.False(System.Runtime.Intrinsics.X86.Avx.IsSupported);
+                Assert.True(System.Runtime.Intrinsics.X86.Sse.IsSupported);
+            }
+            else if (mode == disableAvxAndSse)
+            {
+                Assert.True(System.Runtime.Intrinsics.X86.Avx.IsSupported);
```

Comment:
I assume this deserves a comment and a link to an issue...

Reply:
Even if there was an issue, I would still expect this to be testing `Assert.False` and for us to have an issue linked to whatever workaround we are using instead (which would likely be explicitly setting these both to false).

Reply:
I tried with the recent version of the core-sdk and the issue is still there. i.e setting COMPlus_EnableHWIntrinsic=0  is not disabling the avx and sse

Reply:
I'll check locally (on a CoreCLR test) and see if I can spot the problem.